src/encoded/types/shared_calculated_properties.py

```diff
@@ -514,3 +514,20 @@ def replication_type(self, request, replicates=None, assay_term_name=None):
                 return 'isogenic'
 
         return 'anisogenic'
+
+
+class CalculatedBiologicalReplicates:
+    @calculated_property(schema={
+        "title": "Formatted biological replicates",
+        "description": "The biological replicate numbers associated with this dataset as formatted text.",
+        "comment": "Do not submit. This field is calculated through the experiment replicates.",
+        "type": "string"
+    })
+    def biological_replicates(self, request, replicates=None):
+        replicate_numbers = set()
+        for rep in replicates:
```

Comment:
With `replicates=None` default value I think you could be trying to iterate over None here. Could maybe add `condition=replicates` to calculated property so it only runs if replicates is defined?

Reply:
Oh, _that’s_ what `condition` is for!